graylog2-rest-client/src/main/java/org/graylog2/restclient/models/dashboards/widgets/StackedChartWidget.java

```diff
@@ -0,0 +1,128 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.restclient.models.dashboards.widgets;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.graylog2.restclient.lib.timeranges.RelativeRange;
+import org.graylog2.restclient.lib.timeranges.TimeRange;
+import org.graylog2.restclient.models.dashboards.Dashboard;
+
+import java.util.List;
+import java.util.Map;
+
+public class StackedChartWidget extends ChartWidget {
+
+    private final String renderer;
+    private final String interpolation;
+    private final List<Series> chartSeries;
+
+    public StackedChartWidget(Dashboard dashboard, TimeRange timerange, String description, String streamId, String renderer, String interpolation, String interval, List<Map<String, Object>> series) {
+        this(dashboard, null, description, 0, timerange, streamId, renderer, interpolation, interval, series, null);
+    }
+
+    public StackedChartWidget(Dashboard dashboard, String id, String description, int cacheTime, TimeRange timerange, String streamId, String renderer, String interpolation, String interval, List<Map<String, Object>> series, String creatorUserId) {
+        super(Type.STACKED_CHART, id, description, cacheTime, dashboard, creatorUserId, null, timerange, streamId, interval);
+
+        this.renderer = renderer;
+        this.interpolation = interpolation;
+
+        this.chartSeries = Lists.newArrayList();
+        for (Map<String, Object> aSeries : series) {
+            this.chartSeries.add(Series.fromMap(aSeries));
+        }
+    }
+
+    @Override
+    public Map<String, Object> getConfig() {
+        Map<String, Object> config = Maps.newHashMap();
+        config.putAll(getTimerange().getQueryParams());
+        config.putAll(super.getConfig());
```

Comment:
Shouldn't this be the first addition to the map so that the query parameters and the local config can overwrite settings?